lib/anonymousNaming.js

```diff
@@ -43,42 +49,170 @@ function assignUidToParameterSchemas(doc) {
+          //       x-parser-schema-id: mySchemaVehicleArray
+          //       items:
+          //         - $ref: "#/components/schemas/Car"
+          //         - $ref: "#/components/schemas/Airplane"
+
+          // but if the items are anonymous
+          //   properties:
+          //     vehicle:
+          //       type: array
+          //       x-parser-schema-id: mySchemaVehicleArray <-- collision
+          //       items:
+          //         - type: object
+          //           x-parser-schema-id: mySchemaVehicle
+          //           properties:
+          //             maxSpeed: 
+          //               type: string
+          //         - type: object
+          //           x-parser-schema-id: mySchemaVehicle <-- collision
+          //           properties:
+          //             maxAltitude: 
+          //               type: string
+          //
+          // would not work so we can not give code generator firendly id when `items`is an array.
+          return null;
+        }
+
+        return options.parentId + upperFirst(propertyName);
+      },
+      callback: (subSchema, propOrIndex, callbackType, options) => {
+        if (callbackType === SchemaIteratorCallbackType.END_SCHEMA) {
+          return;
+        }
+
+        if (propOrIndex === null &&
+          options.parentType === 'array' // Special edge case, here we need to inheritance the id to the childs.
+        ) {
+          propOrIndex = options.parentId;
+        }
+
+        if (propOrIndex === null) {
+          return false;
+        }
+
+        if (!(subSchema.type() === 'object' ||
+          subSchema.type() === 'array' ||
+          (subSchema.type() === 'string' && subSchema.enum()) ||
+          !subSchema.type() // oneOf, anyOf, allOf dont have a type.
+        )) {
+          // Simple types dont need generated ids.
+          return false;
+        }
+
+        options.parentId = propOrIndex;
+        options.parentType = subSchema.type();
+        options.level++;
+
+        if (!subSchema.$id()) {
+          if (subSchema.json()[String(xParserSchemaIdLevel)] && subSchema.json()[String(xParserSchemaIdLevel)] < options.level) {
+            // After deref we get a tree, always prefer names of parent root schema definitions.
+
+            // The usage of $ref will lead to having the same schema multiple times in the tree caused by deref.
+            // By walking through the tree, each level deper in the tree gets a higher xParserSchemaIdLevel
+            // So if we find a genereated id with a higher xParserSchemaIdLevel the the current, this id was generated because the current schema was used via $ref as subelement in another schema.
+            // and we prefer the generated id that is clother to a rood node in tree.
+            return false;
+          }
+
+          if (subSchema.type() === 'array') {
+            // You want the shorter on the child, this is used by the code generator to define the class name.
```

Comment:
I'm sorry to say this, but I don't get what you mean by:

> You want the shorter on the child

Not wanting to sound picky, really. I just want to understand this piece of code.

Reply:
If you have the array and the array child. As in the example below this line.
There will be 2 ids generated.
The id with the shorter name you want to have on the array child because this is going to be a class name when code is generated. The array it self is just a class member and will be our of interested when it comes to code generation.

In short: you want shorted speaking names for your generated classes.